HeartbeatLogging/Sources/HeartbeatController.swift

```diff
@@ -108,12 +105,41 @@ public final class HeartbeatController {
     }
 
     // Synchronously gets and returns the stored heartbeats, resetting storage
-    // using the given transform. If the operation threw an error, assume no
-    // heartbeats were retrieved or set.
+    // using the given transform. If the operation throws, assume no
+    // heartbeat(s) were retrieved or set.
     if let heartbeatsBundle = try? storage.getAndSet(using: resetTransform) {
       return heartbeatsBundle.makeHeartbeatsPayload()
     } else {
       return HeartbeatsPayload.emptyPayload
     }
   }
+
+  /// Synchronously flushes the heartbeat for today.
+  ///
+  /// If no heartbeat was logged today, the returned payload is empty.
+  ///
+  /// - Note: This API is thread-safe.
+  /// - Returns: A heartbeats payload for the flushed heartbeat.
+  @discardableResult
+  public func flushHeartbeatFromToday() -> HeartbeatsPayload {
+    let todaysDate = dateProvider()
+    var todaysHeartbeat: Heartbeat?
+
+    storage.readAndWriteSync { heartbeatsBundle in
+      guard var heartbeatsBundle = heartbeatsBundle else {
+        return nil // Storage was empty.
+      }
+
+      todaysHeartbeat = heartbeatsBundle.removeHeartbeat(from: todaysDate)
+
+      return heartbeatsBundle
+    }
+
+    // Note that `todaysHeartbeat` is updated in the above read/write block.
+    guard let todaysHeartbeat = todaysHeartbeat else {
```

Comment:
I think an if - else might be clearer than assigning a var todaysHeartbeat to a let todaysHeartbeat.

Reply:
Reworked in ead3a8c